### PR TITLE
Font Dependencies

### PR DESCRIPTION
**Pull request by eddier**
Moves fonts and icons to npm sources for the respective packages.  Run yarn install and gulp SUI after checking out.

### CODE REVIEW

**Comment by cdaringe** [comment]
**[Storybook Hub](https://hub.getstorybook.io/)** is building storybooks for every commit in this PR.

**Latest Storybook: https://197154f5-9053-4a9d-826d-49820a8322d1.sbook.io/**
<sub>( Visit [here](https://hub.getstorybook.io/app/github.com/Tripwire/octagon/branch/fonts/) for more info )</sub>

> _This comment is automatically added by Storybook Hub_

**Comment by eddier** [comment]
@jhegg can you try this branch out to see if it resolves #23 for you?

**Comment by jhegg** [comment]
Oops, I added a comment about trying this branch to the issue, instead of here. See: https://github.com/Tripwire/octagon/issues/23#issuecomment-283753324

**Comment by jhegg** [comment]
@eddier should the font-awesome artifacts be removed from the style directory? Such as: https://github.com/Tripwire/octagon/blob/9eeb5e34af7a29568af8629d09c7d762cdd7bb94/src/styles/fonts/FontAwesome.otf

**Comment by jhegg** [comment]
One other thing: is it possible to consolidate all of the theme styles/assets inside of the tripwire theme dir? It's weird to see `elegant-icons.css` under `src/styles/`, but we have other icon references inside the semantic stuff. Shouldn't this css be in the override stuff?

**Comment by eddier** [comment]
This is an approach to moving the app.css content to the site.overrides in SUI.  It's just a preliminary push to show the general approach.  I have some concerns about putting all of that content in the theme spec. Primarily that it is blurring the line between what is part of SUI and what is not.  Up to your team but happy to discuss further.

**Comment by jhegg** [comment]
This seems to resolve the missing font issue when I link it into my project, so that's cool! @cdaringe what do you think?

File: scripts/build.js [hunk 1]
```diff
@@ -5,5 +5,7 @@ const build = require('./builder')
 
 build.clean()
 .then(() => build.react())
+.then(() => build.componentcss())
+.then(() => build.componentassets())
```
Comment (cdaringe) [inline scripts/build.js]: nice! can we disambiguate w/ the suir stuff? maybe `octagon` prefix or something?  in the same line of thought, my `react` build step is now equally ambiguous... thatll need some tweaking too :)

File: scripts/builder.js [hunk 2]
```diff
@@ -46,5 +48,17 @@ module.exports = {
     .then(() => mkdirp(this.componentDist))
     .then(() => exec(args.join(' '), { cwd: this.projectRoot, stdio: 'inherit' }))
     .then(([stdout]) => console.log(stdout))
+  },
+  componentcss (opts) {
+    opts = opts || {}
+    const outputDir = this.componentDist + '/styles/components'
```
Comment (cdaringe) [inline scripts/builder.js]: prefer `path.join` for our windoze devs!

File: scripts/builder.js [hunk 3]
```diff
@@ -46,5 +48,17 @@ module.exports = {
...
+    const args = [this.getBin('postcss'), 'src/styles/components/*.css', '-d', outputDir, '-c', this.postCssConfig]
+    return Promise.resolve()
+    .then(() => mkdirp(this.componentDist))
+    .then(() => exec(args.join(' '), { cwd: this.projectRoot, stdio: 'inherit' }))
+    .then(([stdout]) => console.log(stdout))
+  },
+  componentassets (opts) {
+    copy('src/assets', this.assetsDist)
```
Comment (cdaringe) [inline scripts/builder.js]: `return`

**Comment by eddier** [comment]
namespacing applied:

semantic_
octagon_

**Pull request merged**